perfkitbenchmarker/linux_benchmarks/dpb_testdfsio_benchmark.py

```diff
@@ -0,0 +1,162 @@
+"""
+
+import copy
+import datetime
+
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import dpb_service
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.dpb_service import BaseDpbService
+
+BENCHMARK_NAME = 'dpb_testdfsio_benchmark'
+
+BENCHMARK_CONFIG = """
+dpb_testdfsio_benchmark:
+  description: Run testdfsio on dataproc and emr
+  dpb_service:
+    service_type: dataproc
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-4
+          boot_disk_size: 1500
+        AWS:
+          machine_type: m4.xlarge
+      disk_spec:
+        GCP:
+          disk_type: nodisk
+        AWS:
+          disk_size: 1500
+          disk_type: gp2
+    worker_count: 8
+"""
+
+flags.DEFINE_enum('dfsio_fs', BaseDpbService.GCS_FS,
+                  [BaseDpbService.GCS_FS, BaseDpbService.S3_FS,
+                   BaseDpbService.HDFS_FS],
+                  'File System to use in the dfsio operations')
+
+flags.DEFINE_list(
+    'dfsio_file_sizes_list', [1], 'A list of file sizes to use for each of the'
+                                  ' dfsio files.')
+
+flags.DEFINE_list(
+    'dfsio_num_files_list', [10], 'A list of number of dfsio files to use'
+                                  ' during individual runs.')
+
+
+FLAGS = flags.FLAGS
+
+SUPPORTED_DPB_BACKENDS = [dpb_service.DATAPROC, dpb_service.EMR]
+
+
+def GetConfig(user_config):
+    return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites(benchmark_config):
+    """Verifies that the required resources are present.
+
+    Raises:
+      perfkitbenchmarker.errors.Config.InvalidValue: On encountering invalid
+      configuration.
+    """
+    dpb_service_type = benchmark_config.dpb_service.service_type
+    if dpb_service_type not in SUPPORTED_DPB_BACKENDS:
+        raise errors.Config.InvalidValue('Invalid backend for dfsio. Not in:{}'.
+                                         format(str(SUPPORTED_DPB_BACKENDS)))
+
+
+def Prepare(benchmark_spec):
+    del benchmark_spec  # Unused.
+
+
+def Run(benchmark_spec):
+    """Runs testdfsio benchmark and reports the results.
+
+    Args:
+      benchmark_spec: Spec needed to run the testdfsio benchmark
+
+    Returns:
+      A list of samples
+    """
+    run_uri = benchmark_spec.uuid.split('-')[0]
+    source = '/{}'.format(run_uri)
+    update_source_default_fs = False
+
+    if FLAGS.dfsio_fs != BaseDpbService.HDFS_FS:
+        source = '{}:/{}'.format(FLAGS.dfsio_fs, source)
+        benchmark_spec.dpb_service.CreateBucket(source)
+        update_source_default_fs = True
+
+    source_dir = '{}{}'.format(source, '/dfsio')
+
+    combos = ((file_size, num_files)
+              for file_size in FLAGS.dfsio_file_sizes_list
+              for num_files in FLAGS.dfsio_num_files_list)
+
+    results = []
+    for file_size, num_files in combos:
```

Comment:
Suggestion. Instead of pre-computing combos (which is not used elsewhere), just use a nested loop.
https://google.github.io/styleguide/pyguide.html?showone=List_Comprehensions#List_Comprehensions